lib/kitchen/driver/dokken.rb

```diff
@@ -503,8 +525,20 @@ def chef_image
       end
 
       def chef_version
-        return 'latest' if config[:chef_version] == 'stable'
-        config[:chef_version]
+        @chef_version ||= begin
+          provisioner = instance.provisioner
+          if provisioner[:product_name].nil?
+            if config[:chef_version] == 'stable'
+              'latest'
+            else
+              config[:chef_version]
+            end
+          elsif provisioner[:product_name] == 'chef'
+            provisioner[:product_version]
+          else
+            raise "Can only use 'chef' as the product_name with kitchen-dokken"
```

Comment:
This is new behavior. They could specify `chef_image: "chef/chefdk"` or `product_name: "chefdk"` and it would correctly give them a working `chef-client` binary they could use for testing. Does it make sense to have this `raise` in here? I'm leaning towards 'no'.